src/program_generator/pg_arguments.py

```diff
@@ -106,28 +106,41 @@ def _do_generate(self, opcode, count, op_count):
 
     return programs
 
-  # TODO: copied from pg_arythmetic.py. Refactor!
-  def _random_push(self, push):
-    value = random.getrandbits(8*push)
+  def _random_byte_size_push(self, byte_size):
+    value = random.getrandbits(8*byte_size)
+    return self._byte_size_push(byte_size, value)
+
+  def _byte_size_push(self, byte_size, value):
     value = hex(value)
     value = value[2:]
-    if len(value) < 2*push:
-      value = (2*push-len(value))*'0' + value
-    op_num = 6 * 16 + push - 1  # 0x60 is PUSH1
+    if len(value) < 2*byte_size:
+      value = (2*byte_size-len(value))*'0' + value
+    # byte_size is also the OPCODE variant
+    op_num = 6 * 16 + byte_size - 1  # 0x60 is PUSH1
     op = hex(op_num)[2:]
     return op + value
 
   def _generate_program_triplet(self, operation, op_count):
     arity = int(operation['Removed from stack'])
-    arg_bit_sizes = [random.randint(1, 32) for _ in range(0, arity)]
-    single_op_pushes = [self._random_push(size) for size in arg_bit_sizes]
+    opcode = operation['Mnemonic']
+    if opcode in constants.MEMORY_OPCODES:
+      # memory-copying OPCODEs need arguments to indicate up to 64KB of memory
+      args = [random.randint(0, (1<<16) - 1) for _ in range(0, arity)]
```

Comment:
Two generations in row can produce very different program sets: one could copy just 1 byte from the beginning of calldata to the beginning of memory, the other could copy whole 64 bytes.
This is the same for all opcodes with a formula for the cost usage. Maybe we should analyse formulas separately and here we should apply simple arguments so the analysis could be reproduces reliably. That is just my opinion.